### PR TITLE
feat: Adding headers and footers (sketch)

### PR DESCRIPTION
**Pull request by Si-ege**
1. Add navigation and footer

### CODE REVIEW

File: src/components/Footer/index.jsx [hunk 1]
```diff
@@ -0,0 +1,110 @@
+import React from 'react'
+
+
+const App = () => {
+
+
+
+	const footerLinks = [
+		{
```
Comment (guoqqqi) [inline src/components/Footer/index.jsx:9]: please format the code

File: src/components/Footer/index.jsx [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+										<a
+											href={item.href}
+											style={{
+												color: '#4b5563',
+												marginLeft:'0.75rem'
+											}}
+										>{item.icon}</a>
+									)
+
+								})
```
Comment (guoqqqi) [inline src/components/Footer/index.jsx:72]: please setting the spaces to 2

File: src/components/Footer/index.jsx [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+						</span>
+					</div>
+
+				</div>
+			</div>
+
+		</>
+	)
+}
+export default App
```
Comment (guoqqqi) [inline src/components/Footer/index.jsx:110]:
```suggestion
export default App

```
add a blank

File: src/components/Navbar/index.jsx [hunk 4]
```diff
@@ -0,0 +1,100 @@
+import React from "react";
+
+const App = () => {
+	const list = [
+		{
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:5]: ditto

File: src/components/Navbar/index.jsx [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+							})
+						}
+					</div>
+				</div>
+			</div>
+		</>
+	)
+}
+
+export default App
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:100]: ditto

File: src/components/Footer/index.jsx [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+					</div>
+					<div>
+						<span 
+						  style={{
+								fontSize:'12px',
+								color:'#4b5563',
+							}}
+						>
+							Sponsored by 
+							<a href='api7.ai'
```
Comment (guoqqqi) [inline src/components/Footer/index.jsx:83]:
```suggestion
							<a href='https://api7.ai' 
```

File: src/components/Footer/index.jsx [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+							originally built by
+							<a href="https://twitter.com/tim_qian"
+							  style={{
+									textDecoration:'none',
+									backgroundColor:'#60A5FA',
+									padding:'0.25rem 0.5rem',
+									color:'#fff',
+									borderRadius:'1rem 1rem'
+								}}
+							>@tim_qian</a>
```
Comment (guoqqqi) [inline src/components/Footer/index.jsx:100]: remove this

**Comment by guoqqqi** [comment]
image
1. logo and nav item are not centered above and below
2. Replace TODO with Star and the address with the repository address
3. Spacing should be set below the navigation

**Comment by guoqqqi** [comment]
image
IMO we should consider part of the adaptive problem

**Comment by guoqqqi** [comment]
also cc @Baoyuantop PTAL~

**Comment by Baoyuantop** [comment]
Better to run `yarn prettier --write .`

File: src/App.js [hunk 8]
```diff
@@ -356,6 +359,7 @@ const App = () => {
             </TabPanel>
           </div>
         </div>
+        { window.self === window.top ? <Footer /> :  []}
```
Comment (guoqqqi) [inline src/App.js:362]:
```suggestion
        { window.self === window.top ? <Footer /> :  null }
```

File: src/components/Footer/index.jsx (hunk 1 above)
Comment (guoqqqi) [reply src/components/Footer/index.jsx:8]: we should setting the spaces to 2

File: src/components/Footer/index.jsx [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+						justifyContent: "space-between",
+						alignItems: 'center',
+						padding: '0.5rem 0.75rem',
+						borderTop: '1px solid #e5e7eb',
+					}}
+				>
+					<div
+						style={{
+							display: 'flex',
+							alignItems: 'center',
```
Comment (guoqqqi) [inline src/components/Footer/index.jsx:55]: need format the code

File: src/components/Footer/index.jsx [hunk 10]
```diff
@@ -0,0 +1,99 @@
...
+              display: 'flex',
+              alignItems: 'center',
+            }}
+          >
+            {
+              footerLinks.map((item, idx) => {
+                return (
+                  <a
+                    className='leftbottomtext'
+                    key={idx}
```
Comment (guoqqqi) [inline src/components/Footer/index.jsx:62]:
```suggestion
              footerLinks.map((item) => {
                return (
                  <a
                    className='leftbottomtext'
                    key={item.label}
```

File: src/components/Footer/index.jsx [hunk 11]
```diff
@@ -0,0 +1,99 @@
+import React from 'react'
+
+import './style.css'
+
+const App = () => {
+
+  const footerLinks = [
+    {
+      label: '',
```
Comment (guoqqqi) [inline src/components/Footer/index.jsx:9]:
```suggestion
      label: 'API7.ai Logo',
```

File: src/components/Footer/style.css [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+.rightbottomtext:hover {
+  opacity: 0.8;
+}
+
+@media only screen and (max-width:768px) {
+  .footbox {
+		min-width: 90%;
+    flex-direction: column
+  }
```
Comment (guoqqqi) [inline src/components/Footer/style.css:13]: format code

File: src/components/Navbar/index.jsx [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+    {
+      label: 'API Blog',
+      href: 'https://api7.ai/blog',
+    },
+    {
+      label: 'About us',
+      href: 'https://api7.ai/about'
+    }
+  ]
+  const right = [
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:28]:
```suggestion
  const navRightList = [
```

File: src/components/Navbar/index.jsx [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+const App = () => {
+
+  const [open, setOpen] = useState(true)
+
+  const switchBtn = () => {
+    setOpen(open => !open)
+  }
+
+  const list = [
+    {
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:15]:
```suggestion
  const navLeftList = [
    {
```

File: src/components/Navbar/index.jsx [hunk 15]
```diff
@@ -0,0 +1,147 @@
+import React from "react";
+import { useState } from "react";
+
+import './style.css'
+
+const App = () => {
+
+  const [open, setOpen] = useState(true)
+
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:9]:
```suggestion
  const [showNavContent, setShowNavContent] = useState(true)

```

File: src/components/Navbar/index.jsx [hunk 16]
```diff
@@ -0,0 +1,147 @@
...
+  const switchBtn = () => {
+    setOpen(open => !open)
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:11]:
```suggestion
    setOpen(open => !open)
```
ditto

File: src/components/Navbar/index.jsx [hunk 17]
```diff
@@ -0,0 +1,147 @@
...
+    {
+      label: <img alt="Logo" src="https://static.apiseven.com/202108/1648866052600-a71c5469-f81c-4b52-8c20-76eef2987dfc.png" width='120px' height='auto' loading="lazy" class="chakra-image css-1s16w59"></img>,
+      href: '/'
+    },
+    {
+      label: 'API Blog',
+      href: 'https://api7.ai/blog',
+    },
+    {
+      label: 'About us',
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:24]:
```suggestion
      label: 'About Us',
```

File: src/components/Navbar/index.jsx [hunk 18]
```diff
@@ -0,0 +1,147 @@
...
+    setOpen(open => !open)
+  }
+
+  const list = [
+    {
+      label: <img alt="Logo" src="https://static.apiseven.com/202108/1648866052600-a71c5469-f81c-4b52-8c20-76eef2987dfc.png" width='120px' height='auto' loading="lazy" class="chakra-image css-1s16w59"></img>,
+      href: '/'
+    },
+    {
+      label: 'API Blog',
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:20]:
```suggestion
      label: 'Blog',
```

File: src/components/Navbar/index.jsx [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+                return (
+                  <a
+                    className="leftcontent"
+                    href={item.href}
+                    key={idx}
+                  >
+                    {item.label}
+                  </a>
+                )
+              })
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:80]:
```suggestion
              list.map((item) => (
                  <a
                    className="leftcontent"
                    href={item.href}
                    key={item.label}
                  >
                    {item.label}
                  </a>
                ))
```

File: src/components/Navbar/index.jsx [hunk 20]
```diff
@@ -0,0 +1,147 @@
...
+            {
+              right.map((item, key) => {
+                return (
+                  <a
+                    className="rightcontent"
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:89]: `key => item.label`
We generally do not use the index as the key when there are values in the array that are different from each item
https://zh-hans.reactjs.org/docs/lists-and-keys.html#keys

File: src/components/Navbar/index.jsx [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+              })
+            }
+          </div>
+          <div
+            className="switchbtn"
+            onClick={switchBtn}
+          >
+            <span className={open ? "line" : 'line1'}></span>
+            <span className={open ? "line" : 'line2'}></span>
+            <span className={open ? "line" : ''}></span>
```
Comment (guoqqqi) [inline src/components/Navbar/index.jsx:116]: Unfriendly variable names
how about`navMobile` and `navDesktop`

**Review by guoqqqi — COMMENTED** [review]
Good job!

**Comment by guoqqqi** [comment]
LGTM cc @Baoyuantop PTAL~

**Review by juzhiyuan — APPROVED** [review]

**Pull request merged**